bokehjs/src/lib/models/glyphs/webgl/line_gl.ts

```diff
@@ -0,0 +1,184 @@
+import {BaseGLGlyph, Transform} from "./base"
+import {LineView} from "../line"
+import {ReglWrapper} from "./regl_wrap"
+import {color2rgba} from "core/util/color"
+import {resolve_line_dash} from "core/visuals/line"
+import {Texture2D} from "regl"
+
+
+const cap_lookup: {[key: string]: number} = {butt: 0, round: 1, square: 2}
+
+const join_lookup: {[key: string]: number} = {miter: 0, round: 1, bevel: 2}
```

Comment:
It would be better to not provide an explicit type in those two cases. You will get the precise types from type inference, which will not allow you to accidentally provide invalid cases later on.

Reply:
OK.